contracts/src/v0.8/ccip/FeeQuoter.sol

```
@@ -374,6 +370,31 @@ contract FeeQuoter is AuthorizedCallers, IFeeQuoter, ITypeAndVersion, IReceiver,
     return Internal.TimestampedPackedUint224({value: rebasedValue, timestamp: uint32(block.timestamp)});
   }
 
+  /// @dev Gets the fee token price and the gas price, both denominated in dollars.
+  /// @param token The source token to get the price for.
+  /// @param destChainSelector The destination chain to get the gas price for.
+  /// @param destChainConfig The config struct for the dest chain selector.
+  /// @return tokenPrice The price of the feeToken in 1e18 dollars per base unit.
+  /// @return gasPriceValue The price of gas in 1e18 dollars per base unit.
+  function _getTokenAndGasPrices(
```

Getting these two values in a single function was only done because the function was public and therefore saves an RPC call. I don't think this helper function should get the token price as well as they are completely unrelated

---

this helper function is used in `getValidatedFee()` also, where both return values are used

---

yeah but why not have this return just the gasPriceValue and return `((_getValidatedTokenPrice(token), _getTokenAndGasPrices(...)` in the other func? This function does nothing with `token` except pass it to another function

---

oh I see your suggestion now - yes SGTM 👍